src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.cs

```diff
@@ -1,256 +1,124 @@
-                {
-                    return;
-                }
-
-                context.RegisterRefactoring(
-                    new MyCodeAction(Title,
-                        c => UpdateDocumentAsync(root, document, ifStatement, switchSections)),
-                    ifStatement.Span);
+                return;
             }
 
-            private IEnumerable<(IEnumerable<IPattern<TSwitchLabelSyntax>> patterns, TStatementSyntax statement)> GetSections(
-                TIfStatementSyntax rootIfStatement)
+            var index = operations.IndexOf(ifOperation);
+            if (index == -1)
             {
-                // Iterate over subsequent if-statements whose endpoint is unreachable.
-                foreach (var statement in GetSubsequentStatements(rootIfStatement))
-                {
-                    if (!(statement is TIfStatementSyntax ifStatement))
-                    {
-                        yield break;
-                    }
-
-                    if (!CanConvertIfToSwitch(ifStatement))
-                    {
-                        yield break;
-                    }
-
-                    var sectionList = new List<(IEnumerable<IPattern<TSwitchLabelSyntax>>, TStatementSyntax)>();
-
-                    // Iterate over if-else statement chain.
-                    foreach (var (condition, body) in GetIfElseStatementChain(ifStatement))
-                    {
-                        // If there is no condition, we have reached the "else" part.
-                        if (condition == null)
-                        {
-                            _switchDefaultBodyOpt = body;
-                            break;
-                        }
-
-                        var patternList = new List<IPattern<TSwitchLabelSyntax>>();
-
-                        // Iterate over "||" or "OrElse" operands to make a case label per each condition.
-                        var patterns = GetLogicalOrOperands(condition).Reverse().Select(CreatePatternFromExpression);
-                        foreach (var pattern in patterns)
-                        {
-                            // If we could not create a pattern from the condition, we stop.
-                            if (pattern == null)
-                            {
-                                yield break;
-                            }
-
-                            patternList.Add(pattern);
-                        }
-
-                        sectionList.Add((patternList, body));
-                    }
-
-                    foreach (var section in sectionList)
-                    {
-                        yield return section;
-                    }
-
-                    _numberOfSubsequentIfStatementsToRemove++;
-
-                    if (_switchDefaultBodyOpt.HasValue || EndPointIsReachable(ifStatement))
-                    {
-                        yield break;
-                    }
-                }
+                return;
```

Comment:
this would be some sort of bug in the operation model right?  you're just doing this to be hyper sure nothing wonky is going on, right?